### DISCUSSION TITLE
Failed to initialize storage module

### DISCUSSION THREAD

**Discussion by DorianHawkmoon** [daledavies/jump#83] [Q&A] [answered]
Hi! this looks beautiful and as simple as I'd like to, so I'm trying to install it in ArchLinux.
I'm stuck with an error so maybe anyone can help or give a hint here.

Basically I cloned the repo, got latest release 1.3.2 (also tried with latest), copied jumpapp content directory to the webroot, edit config.php and ran composer install --no-dev for the dependencies. I also created the cache folder.
For testing purposes, I'm reusing the folder/nginx/certificate of another website and changed the permission of the folder to 777. 

Whenever I try to access the page, the error that appears on the logs.
`[error] 411#411: *1 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Nette\InvalidStateException: Failed to initialize storage module: files (path: ) in /srv/http/ookami/vendor/nette/http/src/Http/Session.php:119
Stack trace:
#0 /srv/http/ookami/vendor/nette/utils/src/Utils/Callback.php(34): Nette\Http\Session->Nette\Http\{closure}()
#1 [internal function]: Nette\Utils\Callback::Nette\Utils\{closure}()
#2 /srv/http/ookami/vendor/nette/utils/src/Utils/Callback.php(43): session_start()
#3 /srv/http/ookami/vendor/nette/http/src/Http/Session.php(115): Nette\Utils\Callback::invokeSafe()
#4 /srv/http/ookami/vendor/nette/http/src/Http/Session.php(152): Nette\Http\Session->doStart()
#5 /srv/http/ookami/vendor/nette/http/src/Http/SessionSection.php(61): Nette\Http\Session->autoStart()
#6 /srv/http/ookami/classes/Pages/AbstractPage.php(46): Nette\Http\SessionSection->set()
#7 /srv/http/ookami/classes/Main.php(68): Jump\Pages\AbstractPage->__construct()
#8 /srv/http/ookami/index.php(22): Jump\Main->init()
#9 {main}
  thrown in /srv/http" while reading response header from upstream, client: 192.168.1.168, server: ookami.es, request: "GET / HTTP/2.0", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "ookami.es"
`

This is the following nginx configuration  

server {
    server_name ookami.es;
    listen 80;
    include conf.d/commonConfig.conf;
    return 301 https://ookami.es$request_uri;
}
server {
    server_name ookami.es;
    include conf.d/certbot.conf;
    include conf.d/commonConfig.conf;
    root /srv/http/ookami;
    index index.php index.html;
    location / {
	 ssi on;
         ssi_min_file_chunk 1k;
         ssi_silent_errors off;
         ssi_value_length 256;
	 try_files $uri $uri/ index.php$is_args$args;
       }
	access_log /var/log/nginx/ookami.access.log;
	error_log /var/log/nginx/ookami.error.log;
	location ~ \.php$ {
	    try_files $uri =404;
	    fastcgi_pass unix:/run/php-fpm/php-fpm.sock;
	    include fastcgi.conf;
	    fastcgi_param HTTPS on;
	    fastcgi_index $document_root/index.php;
            fastcgi_split_path_info ^((?U).+\.php)(/?.+)$;
            fastcgi_param SCRIPT_FILENAME $document_root/$fastcgi_script_name;
            fastcgi_param PATH_TRANSLATED $document_root/$fastcgi_path_info;
            fastcgi_param PATH_INFO $fastcgi_path_info;
	}
}

**Comment by daledavies** [comment, marked as answer]
Thanks for reaching out, I glad you like Jump 😃

I'm not at my desk at the moment so can't research properly but the error seems to be PHP telling us that is isn't able to save the session state to a file. Can I see your php config for session storage? Also what version of PHP do you have etc.

**Reply by DorianHawkmoon** [reply]
Thanks!

I'm using PHP 8.2.4 and the configuration for session storage is something I haven't touched, so this is what I have out of the box under the session section.
I'm wondering as well the `storage module: files (path: ) ` that appears empty in that message. My logic tells me it should have something. 

[Session]
session.save_handler = files

;session.save_path = "/tmp"

session.use_strict_mode = 0
session.use_cookies = 1

;session.cookie_secure =

session.use_only_cookies = 1
session.name = PHPSESSID
session.auto_start = 0
session.cookie_lifetime = 0
session.cookie_path = /
session.cookie_domain =
session.cookie_httponly =
session.cookie_samesite =
session.serialize_handler = php
session.gc_probability = 1

session.gc_divisor = 1000
session.gc_maxlifetime = 1440
session.referer_check =
session.cache_limiter = nocache
session.cache_expire = 180
session.use_trans_sid = 0

session.sid_length = 26
session.trans_sid_tags = "a=href,area=href,frame=src,form="

;session.trans_sid_hosts=""

session.sid_bits_per_character = 5

;session.upload_progress.enabled = On
;session.upload_progress.cleanup = On
;session.upload_progress.prefix = "upload_progress_"
;session.upload_progress.name = "PHP_SESSION_UPLOAD_PROGRESS"
;session.upload_progress.freq =  "1%"
;session.upload_progress.min_freq = "1"
;session.lazy_write = On

**Reply by DorianHawkmoon** [reply]
Ok, I got it, I had to uncomment the session.save_path because by default is empty, not having a default value ^^U
I'm now investigating the next error

`"PHP` message: PHP Warning:  fopen(/srv/http/ookami/cache/application/_sites%2Fdefault/_66ecfeb9791a023150773849f1550c5d): Failed to open stream: No such file or directory in /srv/http/ookami/vendor/nette/caching/src/Caching/Storages/FileStorage.php on line 128; PHP message: PHP Fatal error:  Uncaught Jump\Exceptions\ConfigException: Config key does not exist... (dockerproxyurl) in /srv/http/ookami/classes/Config.php:118 => Solved too by adding the dockerproxyurl key I was missing. Everything working!!

Which, by the way I had to create folder cache/application before, as it was complaining about not finding that folder. Not sure if that should have been created by jump itself.
